### DISCUSSION TITLE
[Resolved] Can't connect to mail server: imaplib.IMAP4.error: Authentication failed.

### DISCUSSION THREAD

**Discussion by latchee** [jonaswinkler/paperless-ng#1157] [Support]
Hi there,

I have been trying to setup the mail consumer but what ever option I try, it results in connection failure with the error:
imaplib.IMAP4.error: b'[AUTHENTICATIONFAILED] Authentication failed.

My current troublesooting
1. I have tested the mail account (username and password) in thunderbird, it works fine.
2. I have tried STARTTLS and no encryption security over port 143 (also works fine on another email client)
3. I have tried SSL security over 993 (also works fine on another email client)
4. I have change email server to its IP
5. I have put the username and password in ""

None of those are working. Mail server is a simple selfhosted postfix/dovecot with auth_mechanisms = plain (I have tried also auth mechanisms login digest-md5 cram-md5 ntlm rpa apop anonymous but none worked).

I have also tried to install package python3-imaplib2, still  no changes.

Here is the error I see in logs:

> [2021-07-06 11:16:47,191] [ERROR] [paperless.mail.tasks] Error while processing mail account scanner
> Traceback (most recent call last):
>   File "/opt/paperless/src/paperless_mail/mail.py", line 162, in handle_mail_account
>     M.login(account.username, account.password)
>   File "/opt/paperless/.local/lib/python3.8/site-packages/imap_tools/mailbox.py", line 40, in login
>     login_result = self.box.login(username, password)
>   File "/usr/lib/python3.8/imaplib.py", line 603, in login
>     raise self.error(dat[-1])
> imaplib.IMAP4.error: b'[AUTHENTICATIONFAILED] Authentication failed.'
> 
> During handling of the above exception, another exception occurred:
> 
> Traceback (most recent call last):
>   File "/opt/paperless/src/paperless_mail/tasks.py", line 14, in process_mail_accounts
>     total_new_documents += MailAccountHandler().handle_mail_account(
>   File "/opt/paperless/src/paperless_mail/mail.py", line 164, in handle_mail_account
>     raise MailError(
> paperless_mail.mail.MailError: Error while authenticating account scanner

Would you be able to help me with this issue?

Kind regards,

**Comment by budda85** [comment]
Do you have any special characters in your password?  I once had this problem somewhere else and the password was simply not transferred completely.

**Reply by latchee** [reply]
Hi,
Thank you for help. I verified the password and issue was not this. 
But, after all, issue was the way I configured my ldap queries for dovecot to provide content of a mailbox... Authentication was in fact working but dovecot was unable to provide any content for services account. After I fixed this, email consumption started to work!
